src/test/java/edu/harvard/iq/dataverse/api/HarvestingServerIT.java

```diff
@@ -92,4 +103,88 @@ public void testSetCreation()
 		// TODO - get an answer to the question of if it's worth cleaning up (users, sets) or not
 
 	}
+        
+    // A more elaborate test - we'll create and publish a dataset, then create an
+    // OAI set with that one dataset, and attempt to retrieve the OAI record
+    // with GetRecord. 
+    @Test
+    public void testGetRecord() throws InterruptedException {
+
+        setupUsers();
+
+        // create dataverse:
+        
+        Response createDataverseResponse = UtilIT.createRandomDataverse(adminUserAPIKey);
+        createDataverseResponse.prettyPrint();
+        String dataverseAlias = UtilIT.getAliasFromResponse(createDataverseResponse);
+
+        // publish dataverse:
+        
+        Response publishDataverse = UtilIT.publishDataverseViaNativeApi(dataverseAlias, adminUserAPIKey);
+        assertEquals(OK.getStatusCode(), publishDataverse.getStatusCode());
+        
+        // create dataset: 
+        
+        Response createDatasetResponse = UtilIT.createRandomDatasetViaNativeApi(dataverseAlias, adminUserAPIKey);
+        createDatasetResponse.prettyPrint();
+        Integer datasetId = UtilIT.getDatasetIdFromResponse(createDatasetResponse);
+        
+        // retrieve the global id: 
+        String datasetPersistentId = UtilIT.getPersistentDatasetIdFromResponse(createDatasetResponse);
+        
+        // publish dataset:
+        
+        Response publishDataset = UtilIT.publishDatasetViaNativeApi(datasetPersistentId, "major", adminUserAPIKey);
+        assertEquals(200, publishDataset.getStatusCode());
+        
+        String identifier = datasetPersistentId.substring(datasetPersistentId.lastIndexOf('/')+1);
+        
+        logger.info("identifier: "+identifier);
+        
+        // Let's try and create an OAI set with the dataset we have just 
+        // created and published: 
+        
+        String setName = identifier;
+        String setQuery="dsPersistentId:"+identifier;
+        String apiPath = String.format("/api/harvest/server/oaisets/%s",setName);
+                
+                
+        Response createSetResponse = given()
+                .header(UtilIT.API_TOKEN_HTTP_HEADER, adminUserAPIKey)
+                .body(jsonForTestSpec(setName, setQuery))
+                .post(apiPath);
+        assertEquals(201, createSetResponse.getStatusCode());
+        
+        // TODO: a) look up the set via native harvest/server api; 
+        //       b) look up the set via the OAI ListSets;
+        
+        // export set: 
+        // (this is asynchronous - so we should probably wait a little)
+        
+        apiPath = String.format("/api/admin/metadata/exportOAI/%s", setName);
+        Response exportSetResponse = given()
+                .put(apiPath);
```

Comment:
It would be nice to refactor this PUT out of this specific test and into UtilIT.java so it can be reused.